src/vs/workbench/browser/parts/editor/titleControl.ts

```diff
@@ -462,6 +471,10 @@ export abstract class TitleControl implements ITitleAreaControl {
 			actions.push(new Separator(), this.pinEditorAction);
 		}
 
+		const titleBarMenu = this.menuService.createMenu(MenuId.EditorTab, this.contextKeyService);
```

Comment:
Since you update the context key you don't need to create/dispose the menu as it will update when the context keys change.